pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion.go

```diff
@@ -826,8 +826,13 @@ func (t *Test) BackgroundCommand(
 // if passed, is the command run to initialize the workload; it is run
 // synchronously as a regular startup function. `runCmd` is the
 // command to actually run the command; it is run in the background.
+//
+// If overrideBinary is true, the binary used to run the command(s) will
+// be replaced with the cockroach binary of the current version the
+// cluster is running in.
+// TODO(testeng): Replace with https://github.com/cockroachdb/cockroach/issues/147374
 func (t *Test) Workload(
-	name string, node option.NodeListOption, initCmd, runCmd *roachtestutil.Command,
+	name string, node option.NodeListOption, initCmd, runCmd *roachtestutil.Command, overrideBinary bool,
```

Comment:
I kept this behavior as opt in for now to reduce surface area of this change, but I wonder if after #147374 we can make it the default.